crates/bevy_core_pipeline/src/core_3d/mod.rs

```diff
@@ -191,22 +193,57 @@ pub struct Opaque3d {
 /// Data that must be identical in order to batch meshes together.
 #[derive(Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
 pub struct Opaque3dBinKey {
+    /// Various flags, with the [`MeshSlabHash`] in the top bits.
+    ///
+    /// We want this to be first to minimize IBO and VBO changes. See the
+    /// comments in [`MeshSlabHash`] for more details.
+    pub flags: MeshCompareFlags,
+
     /// The identifier of the render pipeline.
     pub pipeline: CachedRenderPipelineId,
 
     /// The function used to draw.
     pub draw_function: DrawFunctionId,
 
-    /// The mesh.
-    pub asset_id: AssetId<Mesh>,
-
     /// The ID of a bind group specific to the material.
     ///
     /// In the case of PBR, this is the `MaterialBindGroupId`.
     pub material_bind_group_id: Option<BindGroupId>,
 
-    /// The lightmap, if present.
-    pub lightmap_image: Option<AssetId<Image>>,
+    /// The lightmap, if present; if not present, this is `AssetId::default()`.
+    pub lightmap_image: AssetId<Image>,
+
+    /// The mesh.
+    ///
+    /// Although we don't have multidraw capability yet, we place this at the
+    /// end to maximize multidraw opportunities in the future.
```

Comment:
Why does order matter here? It's not repr(c) or anything.

Reply:
`PartialOrd` compares from first struct field to last struct field. So fields at the top of the struct will generally be placed together more often.

Reply:
TIL, Thanks.